src/compiler/parser/index.js

```diff
@@ -465,14 +465,28 @@ function processOnce (el) {
 
 function processSlot (el) {
   if (el.tag === 'slot') {
-    el.slotName = getBindingAttr(el, 'name')
-    if (process.env.NODE_ENV !== 'production' && el.key) {
-      warn(
-        `\`key\` does not work on <slot> because slots are abstract outlets ` +
-        `and can possibly expand into multiple elements. ` +
-        `Use the key on a wrapping element instead.`
-      )
+    let slotName 
```

Comment:
Why use let if we are not editing slotName more than once here,why not use const ?

Reply:
I was trying to be consistent with the other code but you're right.

Either way there was another PR for this bug so I haven't followed up on this..